### PR TITLE
fix(loader): avoid NPE in FileLineFetcher close method & update HDFS test

### LINKED ISSUE #706: [Bug] testMultiFilesHaveHeader and testFilterPathBySuffix in HDFSLoaderTest throw NPE when --parser-threads is set to > 1.
### Bug Type (问题类型)

None

### Before submit

- [x] I had searched in the [issues](https://github.com/apache/hugegraph-toolchain/issues) and found no similar issues.

### Environment (环境信息)

- Toolchain Version: v1.7.0

### Expected & Actual behavior (期望与实际表现)

When changing the default value of --parser-threads to greater than 1, running unit tests triggers a NullPointerException (NPE) in testMultiFilesHaveHeader and testFilterPathBySuffix within HDFSLoaderTest.

### Vertex/Edge example (问题点 / 边数据举例)

```javascript

```

### Schema [VertexLabel, EdgeLabel, IndexLabel] (元数据结构)

```javascript

```

### PR DESCRIPTION
**Pull request by bajisn-666**
What is the purpose of the change
This PR fixes a `NullPointerException` in `FileLineFetcher.java`. The NPE occurred when the `close()` method was called, but the `reader` was null. This usually happens in multi-threaded environments or when file initialization fails.

Also updated `HDFSLoadTest.java` to use 4 threads to verify the fix in a concurrent scenario.

### Related issues
Fixes #706

### CODE REVIEW

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java [hunk 1]
```diff
@@ -37,20 +37,19 @@
...
 import org.apache.hugegraph.loader.parser.TextLineParser;
 import org.apache.hugegraph.loader.reader.Readable;
+import org.apache.hugegraph.loader.reader.line.Line;
+import org.apache.hugegraph.loader.reader.line.LineFetcher;
+import org.apache.hugegraph.loader.source.file.Compression;
+import org.apache.hugegraph.loader.source.file.FileFormat;
+import org.apache.hugegraph.loader.source.file.FileSource;
 import org.apache.hugegraph.util.E;
 import org.apache.hugegraph.util.Log;
+import org.slf4j.Logger;
```
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java:52]: use style file to avoid code format diff

https://github.com/apache/incubator-hugegraph-toolchain/blob/master/.editorconfig (use it to override original IDE format)

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java [hunk 2]
```diff
@@ -199,6 +202,11 @@ public void skipOffset(Readable readable, long offset) {
 
         try {
             for (long i = 0L; i < offset; i++) {
+                // Fix NPE: check reader again inside loop
+                if (this.reader == null) {
+                    throw new LoadException("Reader is null when skipping offset of file %s",
+                                            readable);
```
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java:208]: hit 100 chars in one line here?

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java
Comment (imbajin) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java:1]: seems the judgement logic in method is not precise & duplicated

maybe we have a better solution for it

File: hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java [hunk 3]
```diff
@@ -158,6 +157,10 @@ public void closeReader() throws IOException {
     @Override
     public Line fetch() throws IOException {
         while (true) {
+            // Fix NPE: check if reader is null before reading
```
Comment (VGalaxies) [inline hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java:160]: **Medium: Missing regression coverage for local parser-concurrency NPE**

`hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java:160`

**Evidence**
- The fix adds a null-reader guard in `fetch()`, and this PR only adds `--parser-threads 4` to two HDFS tests at `HDFSLoadTest.java:69` and `HDFSLoadTest.java:99`. The existing local functional tests still document the same NPE risk and force `--parser-threads 1` at `FileLoadTest.java:1201`, `FileLoadTest.java:1338`, and `FileLoadTest.java:1636`.

**Impact**
- CI can pass without proving the known local multi-file reader/parser concurrency path is fixed, so the NPE can regress while the old FIXME-protected cases remain disabled.

**Requested fix**
- Re-enable at least one existing NPE-marked local functional test with `--parser-threads > 1`, or add a focused regression test that exercises the fixed reader-null close/fetch path.

**Review by VGalaxies — COMMENTED** [review]
**Review summary**

- **Blocking:** no
- **Summary:** One medium non-duplicative finding remains: the PR does not re-enable any of the existing local-file parser-concurrency NPE regression cases.
- **Evidence:**
  - `git diff --unified=0 origin/master...HEAD -- hugegraph-loader/src/main/java/org/apache/hugegraph/loader/reader/file/FileLineFetcher.java hugegraph-loader/src/test/java/org/apache/hugegraph/loader/test/functional/HDFSLoadTest.java`
  - `rg -n "parser-threads|NullPointerException|NPE" hugegraph-loader/src/test/java/org/apache/hugegraph/loader/test/functional/FileLoadTest.java hugegraph-loader/src/test/java/org/apache/hugegraph/loader/test/functional/HDFSLoadTest.java`
